docs/notebooks/word2vec.ipynb

```diff
@@ -1082,8 +1079,5322 @@
+    "\n",
+    "Additional dependencies : \n",
+    "- sklearn\n",
+    "- numpy\n",
+    "- plotly\n",
+    "\n",
+    "The function below can be used to plot the embeddings in an ipython notebook.\n",
+    "It requires the model as the necessary parameter. If you don't have the model, you can load it by\n",
+    "\n",
+    "`model = gensim.models.Word2Vec.load('path/to/model')`\n",
+    "\n",
+    "If you don't want to plot inside a notebook, set the `plot_in_notebook` parameter to `False`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.decomposition import IncrementalPCA    # inital reduction\n",
+    "from sklearn.manifold import TSNE                   # final reduction\n",
+    "import numpy as np                                  # array handling\n",
+    "\n",
+    "from plotly.offline import init_notebook_mode, iplot, plot\n",
+    "import plotly.graph_objs as go\n",
+    "\n",
+    "def reduce_dimensions(model, plot_in_notebook = True):\n",
+    "\n",
+    "    num_dimensions = 2        # final num dimensions (2D, 3D, etc)\n",
+    "\n",
+    "    vectors= []         # positions in vector space\n",
+    "    labels = []         # keep track of words to label our data again later\n",
+    "    for word in model.wv.vocab:\n",
+    "        vectors.append(model[word])\n",
+    "        labels.append(word)\n",
+    "\n",
+    "\n",
+    "    # convert both lists into numpy vectors for reduction\n",
+    "    vectors = np.asarray(vectors)\n",
+    "    labels =  np.asarray(labels)\n",
+    "    \n",
+    "    # reduce using t-SNE\n",
+    "    vectors = np.asarray(vectors)\n",
+    "    logging.info('starting tSNE dimensionality reduction. This may take some time.')\n",
+    "    tsne = TSNE(n_components=num_dimensions, random_state=0)\n",
+    "    vectors = tsne.fit_transform(vectors)\n",
     "\n",
+    "    x_vals = [ v[0] for v in vectors ]\n",
+    "    y_vals = [ v[1] for v in vectors ]        \n",
+    "        \n",
+    "    # Create a trace\n",
+    "    trace = go.Scatter(\n",
+    "        x = x_vals,\n",
+    "        y = y_vals,\n",
+    "        mode = 'text',\n",
+    "        text = labels\n",
+    "        )\n",
+    "    \n",
+    "    data = [trace]\n",
+    "    \n",
+    "    logging.info('All done. Plotting.')\n",
+    "    \n",
+    "    if plot_in_notebook:\n",
+    "        init_notebook_mode(connected=True)\n",
+    "        iplot(data, filename='word-embedding-plot')\n",
+    "    else:\n",
+    "        plot(data, filename='word-embedding-plot.html')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2017-06-22 11:06:03,601 : INFO : starting tSNE dimensionality reduction. This may take some time.\n",
+      "2017-06-22 11:06:11,212 : INFO : All done. Plotting.\n"
```

Comment:
Please add description for "clusters" (why do words fall into one / different clusters?).